pkg/registry/core/pod/strategy.go

```diff
@@ -308,6 +311,18 @@ func dropNonResizeUpdates(newPod, oldPod *api.Pod) *api.Pod {
 		pod.Spec.Containers[idx].Resources = ctr.Resources
 		pod.Spec.Containers[idx].ResizePolicy = ctr.ResizePolicy
 	}
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.SidecarContainers) {
+		for _, ctr := range newPod.Spec.InitContainers {
+			idx, ok := oldInitCtrToIndex[ctr.Name]
+			if !ok {
+				continue
+			}
+			pod.Spec.InitContainers[idx].Resources = ctr.Resources
+			pod.Spec.InitContainers[idx].ResizePolicy = ctr.ResizePolicy
```

Comment:
Good question. IMO we should fail validation, but I don't fully understand the motivation for silently ignoring changes generally.

Reply:
With my implementation, it will reject resize for non-restartable init containers. If you will see `ValidatePodResize` func, I am not dropping CPUMEM resources for non-restartable init containers, thus validation will fail if user will attempt to change resources for non-restartable init containers.

Reply:
We will see the error message like this:

`# pods "..." was not valid:
# * spec: Forbidden: only cpu and memory resources are mutable`

Reply:
> We will see the error message like this:
> 
> "pods "..." was not valid:
> 
> * spec: Forbidden: only cpu and memory resources are mutable"

This error is a bit misleading. It would be good to return an error like

```
only sidecar container resources are mutable.
````

Reply:
Agree with fail